TODO

```diff
@@ -0,0 +1,29 @@
+- Set permissions on the named pipe/Unix socket correctly. No need to let
+  random users spy on worktrees' activity to which they have no read access.
+
+- Skip reporting directories (meaning: make sure we know the type of a path)
+  - To do that, be smart and look whether a path refers to a file in a
+    subdirectory, then make sure that the subdirectory is marked as such
+  - Maybe over-report, and only mark known directories?
+  - Or not worry about this for now, it's okay to report directories, use
+    Trace2 to measure the impact of this
+
+- Exclude submodules somehow.
+
+- Handle when a file is removed and a directory added in its place: this needs
+  to be reported as "deleted".
+
+- Handle when a directory is renamed: only the directory name is reported, but
+  we need to report all paths.
```

Comment:
I could imagine that we're doing the wrong thing here. Do we have tests for that yet?

Reply:
I think we get FS events for the old-name and the new-name on all platforms.  The daemon cannot tell the client that a rename happened, but it should say that both pathnames are dirty.  The client should, for example, drop untracked-cache entries for the old path and create new ones for the new path (as opposed to repurposing the existing objects).
So I think we're good, but I'll add a task to create unit tests to confirm this.